prov/efa/src/efa_mr.c

```diff
@@ -589,6 +589,66 @@ int efa_mr_is_cuda_memory_freed(struct efa_mr *efa_mr, bool *freed)
 	return -FI_EINVAL;
 }
 
+static struct fi_mr_attr *
+dup_mr_attr(const struct fi_mr_attr *attr, uint64_t flags)
+{
+	struct fi_mr_attr *dup_attr;
+
+	dup_attr = calloc(1, sizeof(*attr) +
+			     sizeof(*attr->mr_iov) * attr->iov_count);
+	if (!dup_attr)
+		return NULL;
+
+	*dup_attr = *attr;
+	dup_attr->mr_iov = (struct iovec *) (dup_attr + 1);
+
+	/*
+	 * dup_mr_attr is only used insided ofi_mr_map_insert.
+	 * dmabuf must be converted to iov before the attr
+	 * is inserted to the mr_map
+	 */
+	if (flags & FI_MR_DMABUF)
+		ofi_mr_get_iov_from_dmabuf((struct iovec *)dup_attr->mr_iov,
+			attr->dmabuf, attr->iov_count);
+	else
+		memcpy((void *) dup_attr->mr_iov, attr->mr_iov,
+			sizeof(*attr->mr_iov) * attr->iov_count);
+
+	return dup_attr;
+}
+
+static int efa_mr_map_insert(struct ofi_mr_map *map,
```

Comment:
Could you add a comment on the difference between `efa_mr_map_insert` and `ofi_mr_map_insert`?